lib/acme/client/certificate_request.rb

```diff
@@ -79,7 +79,7 @@ def validate_subject_common_name
 
   def generate
     OpenSSL::X509::Request.new.tap do |csr|
-      csr.public_key = @private_key.public_key
```

Comment:
This is a pretty weird change to be honest. Doesn't `OpenSSL::PKey::EC` respond to public key?

I understand that the private key would work here but I'm concerned to passing down the stack a private key when its not really needed.